### DISCUSSION TITLE
[React] 怎么更新组件props上的属性？

### DISCUSSION THREAD

**Discussion by ytftianwen** [alibaba/formily#3244] [Q&A 2.x] [answered]
问题：怎么更新组件中的props属性
尝试：外部控制`x-component-props`中的`bordered`属性来控制更新组件的props
结果：更新`x-component-props`中的属性未更新组件的`bordered`属性

在线地址：[复现demo](https://codesandbox.io/s/romantic-sid-xq9svs?file=/App.tsx)

大佬们这种情况有什么方法处理呢

**Comment by janryWang** [comment, marked as answer]
https://formilyjs.org/zh-CN/guide/advanced/controlled

**Reply by ytftianwen** [reply]
嗯嗯看到了呢，不过这个写法看起来不太优雅
`x-reactions={(field) => {
              field.component[1].placeholder = obs.input || '受控者'
            }}`
            `field.component[1]`这个取值
            @janryWang 大佬有其它推荐的写法不

**Comment by zFitness** [comment]
你好，试试这种？
```tsx

export default () => {
  const [bordered, setBorder] = useState(true);
  const form = useMemo(() => {
    return createForm();
  }, [bordered]);
}

```

触发表单重新渲染

**Reply by ytftianwen** [reply]
这样的话，已经输入的值需要重新设置给form
找到一种写法
1. 先创建一个observable的变量
2. 在`x-reactions`中调用`field.setComponentProps`
<img width="795" alt="image" src="https://user-images.githubusercontent.com/13484089/179473048-ab981e70-f84e-4e10-807b-c1efa39269a5.png">

**Reply by Ryuurock** [reply]
这个问题真的，解决起来一点都不优雅

**Reply by CandenWang** [reply]
你好，求问还需要observer包装一层吗，有具体的demo链接吗
